site2/docs/concepts-messaging.md

```diff
@@ -617,6 +649,22 @@ Producer<byte[]> producer = client.newProducer()
                 .create();
 ```
 
+## Message redelivery
+
+Apache Pulsar supports graceful failure handling and ensures critical data is not lost. Software will always have unexpected conditions and at times messages may not be delivered successfully. Therefore, it is important to have a built-in mechanism that handles failure, particularly in asynchronous messaging as highlighted in the following examples.
+
+- Consumers get disconnected from the database or the HTTP server. When this happens, the database is temporarily offline while the consumer is writing the data to it and the external HTTP server that the consumer calls is momentarily unavailable.
```

Comment:
```suggestion
- Consumers get disconnected from the database or the HTTP server. When this happens, the database is temporarily offline while the consumer is writing the data to it and the external HTTP server that the consumer calls are momentarily unavailable.
```

Reply:
Thanks for all your comments. But here I think the subject is `the external HTTP server`, so no modification is needed.

Reply:
Oops, my bad.